src/Nox.Lib/Middlewares/RelatedEndpointsMiddleware.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Microsoft.AspNetCore.Http;
+using Nox.Lib;
+using Nox.Solution;
+
+namespace Nox.Middlewares;
+
+internal class RelatedEndpointsMiddleware
+{
+    private readonly PathString _apiPrefix;
+
+    private readonly RequestDelegate _next;
+
+    private readonly IEnumerable<string> _entitiesPluralNames;
+
+    private readonly IEnumerable<string> _navigationNames;
+
+    public RelatedEndpointsMiddleware(RequestDelegate next, NoxSolution solution)
+    {
+        _next = next;
+
+        _apiPrefix = solution.Presentation.ApiConfiguration.ApiRoutePrefix;
+
+        if (solution.Domain is null)
+        {
+            _entitiesPluralNames = Array.Empty<string>();
+            _navigationNames = Array.Empty<string>();
+        }
+        else
+        {
+            _entitiesPluralNames = solution.Domain.Entities.Select(e => e.PluralName).ToList();
+            _navigationNames = solution.Domain.Entities
```

Comment:
we are using this for search consider using hashset instead

Reply:
updated